packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -1269,7 +1286,13 @@ class CupertinoNavigationBarBackButton extends StatelessWidget {
         ),
       ),
       padding: EdgeInsets.zero,
-      onPressed: () { Navigator.maybePop(context); },
+      onPressed: () {
+        if (onPressed != null ){
```

Comment:
I think it might be nice though we should also stay consistent with the other button presses which have no arguments in their onPressed. 

You can pop to root generally via Navigator.of(context, rootNavigator: true).